assets/src/edit-story/components/canvas/multiSelectionMovable.js

```diff
@@ -22,23 +23,113 @@ function MultiSelectionMovable( { selectedElements, nodesById } ) {
 		if ( moveable.current ) {
 			moveable.current.updateRect();
 		}
-	} );
+	}, [ selectedElements, moveable, nodesById ] );
+
+	const { actions: { updateElementsById } } = useStory();
 
-	const targets = selectedElements.map( ( element ) => nodesById[ element.id ] );
+	// Create targets list including nodes and also necessary attributes.
+	const targetList = selectedElements.map( ( element ) => {
+		return {
+			ref: nodesById[ element.id ],
+			id: element.id,
+			x: element.x,
+			y: element.y,
+			rotationAngle: element.rotationAngle,
+		};
+	} );
 	// Not all targets have been defined yet.
-	if ( targets.includes( undefined ) ) {
+	if ( targetList.some( ( { ref } ) => ref === undefined ) ) {
 		return null;
 	}
+
+	/**
+	 * Set style to the element.
+	 *
+	 * @param {Object} target Target element to update.
+	 * @param {Object} frame Properties from the frame for that specific element.
+	 */
+	const setTransformStyle = ( target, frame ) => {
+		target.style.transform = `translate(${ frame.translate[ 0 ] }px, ${ frame.translate[ 1 ] }px) rotate(${ frame.rotate }deg)`;
+	};
+
+	const frames = targetList ? targetList.map( ( target ) => ( {
+		translate: [ 0, 0 ],
+		rotate: target.rotationAngle,
+	} ) ) : [];
+
+	/**
+	 * Resets Movable once the action is done, sets the initial values.
+	 */
+	const resetMoveable = () => {
+		targetList.forEach( ( { ref }, i ) => {
+			frames[ i ].translate = [ 0, 0 ];
+			ref.style.transform = '';
+			ref.style.width = '';
+			ref.style.height = '';
+		} );
+		if ( moveable.current ) {
+			moveable.current.updateRect();
+		}
+	};
+
 	return (
 		<Movable
 			ref={ moveable }
 			zIndex={ 0 }
-			targets={ targets }
+			target={ targetList.map( ( { ref } ) => ref ) }
 
-			// todo@: implement group transformations.
+			// @todo: implement group resizing.
 			draggable={ true }
-			resizable={ true }
+			resizable={ false }
 			rotatable={ true }
+
+			onDragGroup={ ( { events } ) => {
+				events.forEach( ( { target, beforeTranslate }, i ) => {
+					const sFrame = frames[ i ];
+					sFrame.translate = beforeTranslate;
+					setTransformStyle( target, sFrame );
+				} );
+			} }
+			onDragGroupStart={ ( { events } ) => {
+				events.forEach( ( ev, i ) => {
```

Comment:
This and logic `onRotateGroupStart` are very similar. Can these be extracted to a function?

Reply:
Thanks for noting that, created a function for this and also for ending the event, see 0c1d528.